lib/tapioca/generator.rb

```diff
@@ -522,5 +543,32 @@ def compile_dsl_rbi(constant, contents)
 
       filename
     end
+
+    sig { params(tmp_dir: Pathname).returns(T.nilable(String)) }
+    def verify_dsl_rbi(tmp_dir:)
+      existing_rbis = get_file_list(dir: config.outpath).sort
+      new_rbis = get_file_list(dir: tmp_dir).sort
+
+      return "New file(s) introduced." if existing_rbis.length != new_rbis.length
+
+      desynced_files = []
+
+      (0..existing_rbis.length - 1).each do |i|
+        desynced_files << new_rbis[i] unless FileUtils.identical?(existing_rbis[i], new_rbis[i])
+      end
+
+      unless desynced_files.empty?
+        filenames = desynced_files.map { |f| f.to_s.sub!(tmp_dir.to_s, "sorbet/rbi/dsl") }.join("\n  - ")
+
+        return "File(s) updated:\n  - #{filenames}"
+      end
+
+      nil
+    end
+
+    sig { params(dir: Pathname).returns(T::Array[Pathname]) }
+    def get_file_list(dir:)
+      Dir.glob("#{dir}/**/*.rbi").reject { |e| e =~ /gems/ }.map { |f| Pathname.new(f) }
```

Comment:
Aren't you also getting the `shims/` like this?

Why not directly taking the `**/dsl/*.rbi` files?

Reply:
Good call, I don't believe the test suite includes any shim tests. But I updated to exclude that directory. Kept it relatively generic because we just dump the temporary RBIs into a tmp dir without the overall `rbi/dsl` structure.